xbmc/filesystem/FileCache.cpp

```diff
@@ -312,25 +349,30 @@ void CFileCache::Process()
       }
     }
 
-    const int64_t maxWrite = m_pCache->GetMaxWriteSize(m_chunkSize);
-    int64_t maxSourceRead = m_chunkSize;
+    const int64_t maxWrite = m_pCache->GetMaxWriteSize(requestSize);
+    // Make maxSourceRead a multiple of the chunk size, bounded by maxWrite
+    // If we are falling behind we will use larger and larger reads which
+    // over network file systems will hopefully increase throughput.
+    // Special case: m_chunkSize == 1 (non-buffered file) fits
+    // right into this calculation.
+    int64_t maxSourceRead = maxWrite - maxWrite % m_chunkSize;
```

Comment:
Please use () for clarity

Reply:
I haven't added them, but the code looks different now anyway, please tell me if you still think it's needed.